src/async_wrap.cc

```diff
@@ -80,6 +80,20 @@ struct AsyncWrapObject : public AsyncWrap {
   inline AsyncWrapObject(Environment* env, Local<Object> object,
                          ProviderType type) : AsyncWrap(env, object, type) {}
 
+  static Local<FunctionTemplate> GetConstructorTemplate(Environment* env) {
+    Local<FunctionTemplate> tmpl = env->async_wrap_object_ctor_template();
+    if (tmpl.IsEmpty()) {
+      tmpl = env->NewFunctionTemplate(AsyncWrapObject::New);
+      tmpl->SetClassName(
+          FIXED_ONE_BYTE_STRING(env->isolate(), "AsyncWrap"));
+      tmpl->Inherit(AsyncWrap::GetConstructorTemplate(env));
+      tmpl->InstanceTemplate()->SetInternalFieldCount(
+          AsyncWrap::kInternalFieldCount);
```

Comment:
```suggestion
          AsyncWrapObject::kInternalFieldCount);
```

```suggestion
      tmpl->Inherit(AsyncWrapObject::GetConstructorTemplate(env));
```

(I know this was present in the previous code already, but we’re generally going for the most specific class here)

Reply:
Thanks for the suggestion. I’ll see about updating this maybe later tonight or tomorrow. 🙂

Reply:
Fixed.